src/game_battler.cpp

```diff
@@ -370,6 +370,9 @@ bool Game_Battler::AddState(int state_id, bool allow_battle_states) {
 	if (GetSignificantRestriction() != lcf::rpg::State::Restriction_normal) {
 		SetIsDefending(false);
 		SetCharged(false);
+		if (GetBattleAlgorithm() != nullptr) {
+			this->SetBattleAlgorithm(std::make_shared<Game_BattleAlgorithm::NoMove>(this));
+		}
```

Comment:
Overall, this is the right approach. 

However, it's not clear whether this should be `NoMove` `Null` or if we just set the algo itself to `nullptr`. Need to test the behavior carefully in 2k and 2k3.

Reply:
At first I tried `nullptr` but this approach throwed `Invalid battle action` error messages.

Reply:
Right, `nullptr` would mean it's up to the battle system to decide what to do, which would require more changes elsewhere.